[GCheeYang] iP

GCheeYang opened the pull request:
# Tom boi


> Tom frees your mind of having to remember things you need to do. It's

* text-based
* easy to learn
* ~FAST~ *SUPER* FAST to use

All you need to do is,

1. download it from [here.](https://github.com/GCheeYang/ip)
1. double-click it.
1. add your tasks.
1. let it manage your tasks for you 😉

And it is **FREE!**

Features:

- [ ] Managing tasks  
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

###

If you are a Java programmer, you can use it to practice Java too. Here's the main method: 

```ruby
public class Main {
   public static void main(String[] args) {
      Application.launch(MainApp.class, args);
   }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Commit:
Level 1-1

Commit:
Revert "Level 1-1"

This reverts commit 8479005f048ffdb3d401e3c9d4a4e96ec39c9a44.

Commit:
Level 1-2

Commit:
Level 1-3

Commit:
Level 1-4

Commit:
Level 1-5

Commit:
Level 1-6

Commit:
Level 1-6

Commit:
Level 1-6

Commit:
Level-7

Commit:
Level-8

Commit:
Merge conflicts

Commit:
Merge Conflicts resolution

huangtian666 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,24 @@
+public class Task {
+    private String description;
+    private boolean status;
```
I like how your code is simple, clean and functional! Just one minor change here to better fit in the coding standard.
status could be change to something like isDone or isCompleted comply to java's coding convention for boolean 😊

huangtian666 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,24 @@
+        return (status ? "X" : " ");
+    }
+
+    public void toggle() {
```
You could consider renaming the method to toggleStatus() to enhance readability and clarity.

huangtian666 commented on src/main/java/ToDo.java:
```diff
@@ -0,0 +1,13 @@
+public class ToDo extends Task{
```
It is nice to see that you have implemented in an OOP way by showing inheritance.

huangtian666 commented on src/main/java/Tom.java:
```diff
@@ -0,0 +1,112 @@
+import java.util.*;
```
Just a minor change again, it could be better if you can import statement by statement!

huangtian666 commented on src/main/java/Tom.java:
```diff
@@ -0,0 +1,112 @@
+
+    }
+
+    private static void addToDo(String input) {
```
I like how you have increased readability by separating each code into different smaller methods!

huangtian666 reviewed (commented):
Overall, your code is well-structured, clean, and easy to read ! Added some minor changes that you could consider making.

broccoli-hater commented on src/main/java/Deadlines.java:
```diff
@@ -0,0 +1,12 @@
+public class Deadlines extends Task{
```
Good work using OOP! Although, should the class name be a plural? I noticed for the 'ToDo' class however the naming is singular, so just checking if that is what you intended.

broccoli-hater commented on src/main/java/Events.java:
```diff
@@ -0,0 +1,14 @@
+public class Events extends Task{
+    private String from;
```
It's good that the scope of the properties is set to private!
Although I would recommend a more descriptive name. For example, 'startTime' and 'endTime' instead.

broccoli-hater commented on src/main/java/Tom.java:
```diff
@@ -0,0 +1,112 @@
+        }
+    }
+
+    private static void addDeadline(String input) {
```
The method names are very descriptive and easy to understand. I think adding some Javadoc comments would help make it even better.

broccoli-hater commented on src/main/java/Tom.java:
```diff
@@ -0,0 +1,112 @@
+
+        System.out.println(line + "Hello! I'm Tom\n" + "What Can I do for you?" + line);
+
+        while(true) {
```
Minor issue but I think there should be a white space before the opening bracket

broccoli-hater commented on src/main/java/Tom.java:
```diff
@@ -0,0 +1,112 @@
+        System.out.println(line + "Hello! I'm Tom\n" + "What Can I do for you?" + line);
+
+        while(true) {
+            String input = scanner.nextLine(); //takes in user input
```
Nice use of comments! It helps make the code easier to read!

broccoli-hater reviewed (commented):
Overall good and descriptive naming conventions except for in some minor cases.
Good job! Would like to see more from you!

GCheeYang committed:
Update README.md

GCheeYang committed:
Update README.md

GCheeYang committed:
Update README.md

GCheeYang committed:
Update README.md

GCheeYang committed:
Update README.md

GCheeYang committed:
Update README.md

GCheeYang committed:
Update README.md

GCheeYang committed:
Update README.md

GCheeYang committed:
Update README.md

Commit:
Merge branch 'branch-Level-7'

Commit:
Level-9

Commit:
Merge branch 'branch-Level-8'

Commit:
Level-9

Commit:
Level-9

Commit:
Merge branch 'branch-Level-10'

Commit:
Merge branch 'master' of https://github.com/GCheeYang/ip

Commit:
Gradle?

Commit:
Merge remote-tracking branch 'origin/add-gradle-support'

Commit:
Level-10

Commit:
Level-10

Commit:
Update

Commit:
screw up files gone :)

Commit:
screw up change

Commit:
screw up change

Commit:
Abstracting commands to seperate classes

As there are several commands that users could use, each command is extracted and created as a new class.

Commit:
Creating GUI

The main.java, dialogbox.java and mainwindow.java are files that help do create the GUI

This improves the user experience and helps the user to view the UI more pleasantly

Commit:
branch-A-Assertions

Commit:
branch-A-Assertions

Commit:
branch-A-CodeQuality

Commit:
A-UserGuide

Commit:
A-UserGuide

Commit:
A-UserGuide

Commit:
mini updates

Commit:
Merge tag 'branch-A-Assertions'

Commit:
Merge branch 'branch-A-CodeQuality'

# Conflicts:
#	src/main/java/Tom/commands/Parser.java

Commit:
Merging PRs

Commit:
Merging PRs test

Commit:
codequality merge

Commit:
Add Java Docs

Adding java docs to all files in Task folder to ensure readability of code.

GCheeYang committed:
Merge pull request #2 from GCheeYang/branch-A-CodeQuality

Add Java docs to all files in Task folder

Commit:
Add bye command

Add a bye command to exit the program after user is done with application.

Commit:
Ensure Tom.java follows SLAP

Abstracting all the methods from main java class Tom.java to commands folder

Ensuring that code in Tom.java follows SLAP guidelines.

Commit:
BCD Extension

Creating a new function help for users to identify commands available

This allows users to fully comprehend what commands they are able to use in Tom

Commit:
Java Docs for commands

Indicating the java docs for methods under commands to ensure readability

Commit:
data

Commit:
Abstract Task Types

Tasks such as todo, deadline and event are categorized under task for better abstraction

This allows different types of tasks to be created and users to interact with different tasks

Commit:
java fx test fix

Commit:
java fx test fix 2

Commit:
java fx test fix 3

Commit:
java fx test fix 4

Commit:
java fx test fix 4

Commit:
java fx test fix 4

Commit:
refactor: Extract commands into separate classes for better maintainability

Each command is now implemented as a separate class, improving code modularity
and maintainability.

Commit:
feat(gui): Implement GUI using main.java, dialogbox.java, and mainwindow.java

Introduced a graphical user interface to improve user experience. This change allows users to interact with the application more effectively through a well-structured UI.

Commit:
docs: Add JavaDocs to all files in Task folder

Added JavaDocs to improve code readability and maintainability.

Now, all methods and classes in the Task folder have detailed documentation
to help future developers understand their functionality.

Commit:
feat: Implement bye command to exit the application

Added a command that allows users to exit the application when they are done.

This enhances the user experience by providing a clear way to terminate the program.

Commit:
refactor: Apply SLAP principles to Tom.java by abstracting methods

Refactored Tom.java to move command-handling methods into separate classes.

This change improves code organization and adheres to the
Single Level of Abstraction Principle (SLAP), making the code more readable.

Commit:
feat: Add command help feature for users (BCD Extension)

Implemented a new function to display available commands

This helps users identify and understand the commands they can use within the application.

Commit:
minor change

Commit:
bug fixes

Commit:
bug fixes 2

Commit:
final

Commit:
refactor: Extract commands into separate classes for better maintainability

Each command is now implemented as a separate class, improving code modularity
and maintainability.

Commit:
feat(gui): Implement GUI using main.java, dialogbox.java, and mainwindow.java

Introduced a graphical user interface to improve user experience.

This change allows users to interact with the application more effectively through a well-structured UI.

Commit:
docs: Add JavaDocs to all files in Task folder

Added JavaDocs to improve code readability and maintainability.

Now, all methods and classes in the Task folder have detailed documentation
to help future developers understand their functionality.

Commit:
feat: Implement bye command to exit the application

Added a command that allows users to exit the application when they are done.

This enhances the user experience by providing a clear way to terminate the program.

Commit:
refactor: Apply SLAP principles to Tom.java by abstracting methods

Refactored Tom.java to move command-handling methods into separate classes.

This change improves code organization and adheres to SLAP, making the code more readable.

Commit:
feat: Add command help feature for users (BCD Extension)

Implemented a new function to display available commands

This helps users identify and understand the commands they can use within the application.

Commit:
refactor: Extract commands into separate classes for better maintainability

Each command is now implemented as a separate class, improving code modularity
and maintainability.

Commit:
feat(gui): Implement GUI using main.java, dialogbox.java, and mainwindow.java

Introduced a graphical user interface to improve user experience.

This change allows users to interact with the application more effectively through a well-structured UI.

Commit:
feat: Implement bye command to exit the application

Added a command that allows users to exit the application when they are done.

This enhances the user experience by providing a clear way to terminate the program.

Commit:
refactor: Apply SLAP principles to Tom.java by abstracting methods

Refactored Tom.java to move command-handling methods into separate classes.

This change improves code organization and adheres to SLAP, making the code more readable.

Commit:
feat: Add command help feature for users (BCD Extension)

Implemented a new function to display available commands

This helps users identify and understand the commands they can use within the application.

GCheeYang committed:
feat: Add command help feature for users (BCD Extension)

Implemented a new function to display available commands

This helps users identify and understand the commands they can use within the application.

GCheeYang committed:
feat: Add command help feature for users (BCD Extension)

Implemented a new function to display available commands

This helps users understand the commands within the application.

GCheeYang committed:
refactor: Apply SLAP principles to Tom.java by abstracting methods
Refactored Tom.java to move command methods into separate classes.

This change improves code organization and adheres to SLAP.

GCheeYang committed:
refactor: Apply SLAP principles to Tom.java by abstracting methods

Refactored Tom.java to move command methods into separate classes.

This change improves code organization and adheres to SLAP.

GCheeYang committed:
feat: Add command help feature for users (BCD Extension)

Implemented a new function to display available commands

This helps users understand the commands within the application.

GCheeYang committed:
feat: Implement bye command to exit the application

Added a command that allows users to exit the application.

This provides a clear way to terminate the program.

GCheeYang committed:
feat(gui): Implement GUI by main.java, dialogbox.java, mainwindow.java

Introduced a graphical user interface to improve user experience.

This allows users to interact with the application more effectively.

GCheeYang committed:
refactor: Extract commands into separate classes

Each command is now implemented as a separate class

GCheeYang committed:
Update README.md

Update README with project details and setup instructions

GCheeYang committed:
A-TextUiTesting

GCheeYang committed:
Junit testing

GCheeYang committed:
Merge branch 'master' of https://github.com/GCheeYang/ip

GCheeYang committed:
A-JUnit

GCheeYang committed:
refactor: Apply SLAP principles to Tom.java by abstracting methods

Refactored Tom.java to move command methods into separate classes.

This change improves code organization and adheres to SLAP.

GCheeYang committed:
feat: Add command help feature for users (BCD Extension)

Implemented a new function to display available commands

This helps users understand the commands within the application.

GCheeYang committed:
feat: Implement bye command to exit the application

Added a command that allows users to exit the application.

This provides a clear way to terminate the program.

GCheeYang committed:
feat(gui): Implement GUI by main.java, dialogbox.java, mainwindow.java

Introduced a graphical user interface to improve user experience.

This allows users to interact with the application more effectively.

GCheeYang committed:
refactor: Extract commands into separate classes

Each command is now implemented as a separate class

GCheeYang committed:
refactor: Apply SLAP principles to Tom.java by abstracting methods

Refactored Tom.java to move command methods into separate classes.

This change improves code organization and adheres to SLAP.
